sound/soc/sof/ipc.c

```diff
@@ -191,6 +187,34 @@ static inline void ipc_log_header(struct device *dev, u8 *text, u32 cmd)
 }
 #endif
 
+/* caller need hold ipc_lock, send next IPC message in list */
+static void ipc_tx_next_msg_unlock(struct snd_sof_ipc *ipc)
+{
+	struct snd_sof_dev *sdev = ipc->sdev;
```

Comment:
So, we either call this ipc_tx_next_msg_unlocked or __ipc_tx_next_msg. The "unlock" at the end suggests that this function will "unlock" something.

Reply:
"unlock" means this function itself will not hold the lock, the caller please hold the ipc lock.